design-proposals/feature-lifecycle.md

```diff
@@ -0,0 +1,317 @@
+# KubeVirt Feature Lifecycle
+
+## Summary
+
+KubeVirt requires a clear policy on how features are introduced,
+evaluated and finally graduated or removed.
+
+This proposal defines the steps and policies to follow in order
+to manage a feature and its lifecycle in KubeVirt.
+
+The proposal is focusing on introducing features in
+a stable API (CRD) version, e.g. `kubevirt.io/v1`.
+
+## Overview
+
+### Motivation
+KubeVirt has grown into a matured virtualization management solution
+with a large set of features.
+
+New features are being proposed and added regularly to its portfolio.
+
+With time, the challenge of supporting and maintaining such a large
+set of features raised the need to re-examine their relevance.
+It also raised the need to examine with more care features graduation.
```

Comment:
This is not only about complete features, however, it is also about figuring out what API works and what doesn't.
There must be a safe way to experiment with the API.